### DISCUSSION TITLE
Allow padding for timeseries

### DISCUSSION THREAD

**Discussion by laurielounge** [grafana/grafana#42687] [Dashboards, Panels & Visualizations]
When one selects a range from the timepicker, the timeseries sets up its x axis according to the timepicker values. This can lead to values being chopped off if the value doesn't fit within the timpicker's range when expressed as UTC, and it can also lead to values sitting right on the extremes of the graph.

Would be handy to be able to pad the left and right of the timeseries by some set value. Like, show an extra day at the left and right. Or an extra hour or whatever. Would give the values room to breathe.
[panel_dt.zip](https://github.com/grafana/grafana/files/7644811/panel_dt.zip)

**Comment by milinpatel13298** [comment]
I would really like the ability to do this too. Currently, I have to manually round times to the next minute/five minutes in order to accomplish this.

Another padding feature that could be helpful would be the ability to pad values when they are null (similar to what's possible in the Alerts reduce expression.

![image](https://github.com/grafana/grafana/assets/25616107/6354a13e-5da9-4b8a-99d5-2e480d6df027)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
